### PR TITLE
Fix Tracing port names in services to comply with Istio convention

### PR DESCRIPTION
**Pull request by mjakobczyk**
**Description**

We should follow [official Istio documentation about port naming convention](https://istio.io/latest/docs/reference/config/analysis/ist0118/). After editing both services there is a similar spec which indicates that the used protocol is TCP.

```yaml
...
spec:
  ...
  ports:
  - name: tcp-metrics
    port: 8383
    protocol: TCP
    targetPort: 8383
  ...
```

Changes proposed in this pull request:

- fix port names in `tracing-jaeger-operator` service to include protocol prefix
- fix port names in `tracing-jaeger-metrics` service to include protocol prefix
- fix port names in `zipkin` service to include protocol prefix
- get rid of the following errors:

```bash
Info [IST0118] (Service tracing-jaeger-operator.kyma-system) Port name metrics (port: 8383, targetPort: 8383) doesn't follow the naming convention of Istio port.
Info [IST0118] (Service tracing-jaeger-metrics.kyma-system) Port name metrics (port: 14269, targetPort: 14269) doesn't follow the naming convention of Istio port.
Info [IST0118] (Service zipkin.kyma-system) Port name jaeger-collector-zipkin (port: 9411, targetPort: 9411) doesn't follow the naming convention of Istio port.
```

**Related issue(s)**

See: https://github.com/kyma-project/kyma/issues/11686

### CODE REVIEW

File: resources/tracing/templates/kyma-additions/metrics-service.yaml [hunk 1]
```diff
@@ -7,7 +7,7 @@ metadata:
     app.kubernetes.io/name: {{ include "jaeger-operator.fullname" . }}-jaeger-metrics
 spec:
   ports:
-  - name: metrics
+  - name: tcp-metrics
```
Comment (chrkl) [inline resources/tracing/templates/kyma-additions/metrics-service.yaml:10]: Guess it's a http connection, right? Iirc correctly, having a http- prefix allows Istio to account proper http response codes in the metrics and visualize in Kiali.

Comment (mjakobczyk) [reply resources/tracing/templates/kyma-additions/metrics-service.yaml:10]: Yes, sure - TCP is simply more generic. If we want to be specific we can call it `http` also. Changing 👍

Comment (chrkl) [reply resources/tracing/templates/kyma-additions/metrics-service.yaml]: Looks like you also have to change the port names in the ServiceMonitors to make the tests happy.

File: resources/tracing/templates/service.yaml [hunk 2]
```diff
@@ -6,7 +6,7 @@ metadata:
 {{ include "jaeger-operator.labels" . | indent 4 }}
 spec:
   ports:
-  - name: metrics
+  - name: tcp-metrics
```
Comment (chrkl) [inline resources/tracing/templates/service.yaml:9]: Let's change this as well

**Comment by chrkl** [comment]
Please test that your PR would not trigger [this](https://github.com/kyma-project/kyma/issues/12977) again. Thanks!

File: resources/tracing/templates/kyma-additions/zipkin-service.yaml [hunk 3]
```diff
@@ -7,7 +7,7 @@ metadata:
     app.kubernetes.io/component: zipkin 
 spec:
   ports:
-  - name: jaeger-collector-zipkin
+  - name: tcp-jaeger-collector-zipkin
```
Comment (chrkl) [inline resources/tracing/templates/kyma-additions/zipkin-service.yaml:10]: We figured out that zipkin is also a http based protocol. Please change this prefix as well.

**Review by chrkl — APPROVED** [review]

**Pull request merged**